addons/delivery/models/sale_order.py

```diff
@@ -37,6 +37,13 @@ def onchange_order_line(self):
         if delivery_line:
             self.recompute_delivery_price = True
 
+    def update_prices(self):
+        """Recompute delivery after pricelist prices reset."""
+        super().update_prices()
+        if any(line.is_delivery for line in self.order_line):
```

Comment:
`if self.delivery_set` the field `delivery_set` do the same compute of `any(line.is_delivery for line in self.order_line)`

Reply:
Thx, will update.